spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-lattice/src/main/java/org/springframework/cloud/data/module/deployer/lattice/ReceptorModuleDeployer.java

```diff
@@ -16,30 +16,105 @@
 
 package org.springframework.cloud.data.module.deployer.lattice;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
+import org.cloudfoundry.receptor.client.ReceptorClient;
+import org.cloudfoundry.receptor.commands.ActualLRPResponse;
+import org.cloudfoundry.receptor.commands.DesiredLRPCreateRequest;
+import org.cloudfoundry.receptor.support.EnvironmentVariable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import org.springframework.cloud.data.core.ModuleDeploymentId;
 import org.springframework.cloud.data.core.ModuleDeploymentRequest;
 import org.springframework.cloud.data.module.ModuleStatus;
 import org.springframework.cloud.data.module.deployer.ModuleDeployer;
+import org.springframework.util.StringUtils;
 
 /**
  * @author Patrick Peralta
  */
 public class ReceptorModuleDeployer implements ModuleDeployer {
+	private static final Logger logger = LoggerFactory.getLogger(ReceptorModuleDeployer.class);
+
+	public static final String DOCKER_PATH = "docker:///springcloud/stream-module-launcher";
+
+	public static final String BASE_ADDRESS = "192.168.11.11.xip.io";
+
+	private final ReceptorClient receptorClient = new ReceptorClient();
+
 	@Override
 	public ModuleDeploymentId deploy(ModuleDeploymentRequest request) {
-		throw new UnsupportedOperationException();
+		ModuleDeploymentId id = ModuleDeploymentId.fromModuleDefinition(request.getDefinition());
+		String guid = guid(id);
+
+		DesiredLRPCreateRequest lrp = new DesiredLRPCreateRequest();
+		lrp.setProcessGuid(guid);
+		lrp.setRootfs(DOCKER_PATH);
+		lrp.runAction().setPath("java");
+		lrp.runAction().addArg("-Djava.security.egd=file:/dev/./urandom");
+		lrp.runAction().addArg("-jar");
+		lrp.runAction().addArg("/module-launcher.jar");
+
+		List<EnvironmentVariable> environmentVariables = new ArrayList<EnvironmentVariable>();
+		Collections.addAll(environmentVariables, lrp.getEnv());
+		environmentVariables.add(new EnvironmentVariable("MODULES", request.getCoordinates().toString()));
+		environmentVariables.add(new EnvironmentVariable("SPRING_PROFILES_ACTIVE", "cloud"));
+
+		lrp.setEnv(environmentVariables.toArray(new EnvironmentVariable[environmentVariables.size()]));
+
+		lrp.setPorts(new int[] {8080, 9000});
```

Comment:
we will need to change the way we handle port mappings for sure - this was taken from the xolpoc-admin originally, and there the idea was to support 8080 (the app itself) as well as 9000 because it would be used by an "http" source module by default. Obviously we need to ensure that the ports that are mapped are the ones being used